src/components/data-management/AddMetadataButton.jsx

```diff
@@ -38,21 +41,23 @@ const AddMetadataButton = ({ samplesTableRef }) => {
   };
 
   const onUploadCellLevelMetadata = async (file) => {
+    setCellLevelUploading(true);
```

Comment:
Why do you track this as a state variable? Isn't it enough that we are tracking it in redux?

Reply:
Initially when the file is being uploaded, it is not in redux so this is an easy way to track it instead of dispatching actions and using redux.